RxJavaTemplate/app/src/androidTest/java/co/nimblehq/rxjava/ui/screens/second/SecondFragmentTest.kt

```diff
@@ -0,0 +1,90 @@
+package co.nimblehq.rxjava.ui.screens.second
+
+import android.Manifest
+import android.app.Activity
+import android.app.Instrumentation
+import android.content.Intent
+import android.os.Bundle
+import android.provider.MediaStore
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.Intents.intending
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasAction
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.rule.ActivityTestRule
+import androidx.test.rule.GrantPermissionRule
+import co.nimblehq.rxjava.R
+import co.nimblehq.rxjava.di.modules.RepositoryModule
+import co.nimblehq.rxjava.domain.test.MockUtil
+import co.nimblehq.rxjava.ui.screens.MainActivity
+import co.nimblehq.rxjava.ui.screens.launchFragment
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@UninstallModules(RepositoryModule::class)
+@HiltAndroidTest
+class SecondFragmentTest {
+
+    @get:Rule(order = 0)
+    var hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    var activityRule = ActivityTestRule(MainActivity::class.java)
+
+    @get:Rule
+    var cameraPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.CAMERA)
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+        launchFragment()
+    }
+
+    @After
+    fun teardown() {
+        Intents.release()
+    }
+
+    @Test
+    fun when_initialize_should_shows_ui_correctly() {
```

Comment:
just like unit test, I think you can add `@Suppress("IllegalIdentifier")` from the beginning to write the function name with `space` instead of `_` no? 🤔

Reply:
It's not working. Naming the test with blank space will result in build failure with this error
`com.android.tools.r8.errors.b: Space characters in SimpleName 'when initialize_should_shows_ui_correctly' are not allowed prior to DEX version 040` 😅